experiment/simulation/js/main.js

```diff
@@ -1 +1,376 @@
-//Your JavaScript goes in here
+document.addEventListener('DOMContentLoaded', function(){
+
+	const playButton = document.getElementById('play');
+	const pauseButton = document.getElementById('pause');
+	const restartButton = document.getElementById('restart');
+	const dampButton = document.getElementById('damping');
+
+	pauseButton.addEventListener('click', function() { window.clearTimeout(tmHandle); });
+	playButton.addEventListener('click', function() {  window.clearTimeout(tmHandle); tmHandle = setTimeout(draw, 1000 / fps); });
+	restartButton.addEventListener('click', function() {restart();});
+	dampButton.addEventListener('click', function() { isdamp = 1-isdamp ; restart();});
+	
+	function init()
+	{
+		rod = [[startx, starty], [endx, starty], [endx, starty + height], [startx, starty + height]];
+
+		ground = [
+			[startx - margin, starty + height + 40],
+			[startx - margin, starty + height],
+			[endx + margin, starty + height],
+			[endx + margin, starty + height + 40],
+		];
+		curr_t = 0;
+	}
+	
+	function restart() 
+	{ 
+		window.clearTimeout(tmHandle); 
+		init();
+		tmHandle = window.setTimeout(draw, 1000 / fps); 
+	}
+
+	const slider_mass = document.getElementById("mass");
+	const output_mass = document.getElementById("demo_mass");
+	output_mass.innerHTML = slider_mass.value; // Display the default slider value
+
+	slider_mass.oninput = function() {
+		output_mass.innerHTML = this.value;
+		mass = Number(document.getElementById("mass").value);
+		radius = mass / 10000;
+		restart();
+	};
```

Comment:
Similar code is repeated for multiple elements. Is it possible to refactor the code into a function and pass the element ids and any extra processing to the function?